st2common/st2common/util/action_db.py

```diff
@@ -195,6 +196,11 @@ def update_liveaction_status(status=None, result=None, context=None, end_timesta
     old_status = liveaction_db.status
     liveaction_db.status = status
 
+    liveaction_db.log.append({
```

Comment:
Yes, I agree.

Ideally we would have a better abstraction (at least a service function or similar), but I don't think DB specific code is too big of a deal right now.

A (DB neutral) alternative is to use locking, but this would require more code.